front/pages/api/v1/w/[wId]/vaults/[vId]/data_source_views/[dsvId]/index.ts

```diff
@@ -149,26 +146,56 @@ export type GetOrPostDataSourceViewsResponseBody = {
 
 async function handler(
   req: NextApiRequest,
-  res: NextApiResponse<
-    WithAPIErrorResponse<
-      GetOrPostDataSourceViewsResponseBody | PatchDataSourceViewResponseBody
-    >
-  >,
+  res: NextApiResponse<WithAPIErrorResponse<DataSourceViewsResponseType>>,
   auth: Authenticator
 ): Promise<void> {
   const dataSourceView = await DataSourceViewResource.fetchById(
     auth,
     req.query.dsvId as string
   );
 
+  if (
```

Comment:
Wait we were not checking this on the Public API 🤔?

Reply:
Technically the dataSourceViewResource as REsourceWithVault should protect us here nonetheless right?